Makefile

```diff
@@ -41,26 +42,37 @@ QEMU = $(shell command -v qemu-system-$(QEMU_ARCH))
 # **********************************
 
 # Compilers/Assemblers/Linkers
-AS      = $(shell command -v i686-elf-as)
 NASM    = $(shell command -v nasm)
-CXX     = $(shell command -v clang++)
-LD      = $(shell command -v ld.lld)
-OBJCP   = $(shell command -v llvm-objcopy)
+AS      = $(shell command -v i686-elf-as)
+CXX     = $(shell command -v i686-elf-gcc)
+LD      = $(shell command -v i686-elf-ld)
+OBJCP   = $(shell command -v i686-elf-objcopy)
 MKGRUB  = $(shell command -v grub-mkrescue)
-# C / C++ flags (include directory)
-CFLAGS :=                   \
-	-I ${SYSROOT}/usr/include/kernel/ \
-	${PANIX_CFLAGS}
-# C++ only flags (-lgcc flag is used b/c it has helpful functions)
-# Flags explained:
-#
-# -Wno-unused-function
-# We need to ignore unused functions because we may use
-# them at a later time. For example, paging disable.
-#
-CXXFLAGS :=                 \
+# Warning flags
+# (Disable unused functions warning)
+WARNINGS :=              \
+	-Wno-unused-function \
+	-Wall                \
+	-Wextra              \
+	-Winline             \
+	-Wshadow             \
+	-Wcast-align         \
+	-Wno-long-long       \
+	-Wpointer-arith      \
+	-Wconversion         \
+	-Wwrite-strings      \
+	-Wredundant-decls    \
+	-Wmissing-declarations
+# TODO: Include -Werror once we
+# Fix all of the warnings
```

Comment:
Have we not fixed all the warnings?

Reply:
Nope, because we added more with these new flags. Most of the ones left are due to implicit casting.

Reply:
Is there any reason they can't be addressed now?

Reply:
I'm working on it. I've already fixed the 90% of other issues but type casting gets tricky, especially when we're dealing with structs which only use 20 bits of an int and stuff. (Yes, it does complain about that).

Reply:
For right now `-Wconversions` has been removed since that was the only flag that was giving us trouble, and I'm not sure if it's something we want to keep. If it is then it can be added back, but we'll have to fix a **ton** of casting warnings